File: cppsnippets/primepairs/primepairs.cpp
/*************************************\
*Just some quick sample C++	     *
*primepairs.cpp : 	 	     *
*generates a list of prime pairs from*
*a minimum value to a maxiumum value *
\************************************/

#include <iostream>
#include <cmath>


using namespace std;

bool isa_prime_num (int num);

int main ()
{

	int range_min, range_max;

	cout << "Enter a minimum value: ";
	cin >> range_min;
	cout << "Enter a maximum value: ";
	cin >> range_max;


	for (int i=range_min; i<range_max ; i++)
	{
		if ((isa_prime_num (i)) && (isa_prime_num (i+2)))
		{
			cout << i << " " << (i+2) << "\n";
		}	
	}

}


//prime_num?: Int -> Boolean
//Is the number a prime number?
bool isa_prime_num (int num)
{
	for (int n=2; n<(num-1); n++)
	{
		if (!(num%n)) return false; 
	}

	return true;
}

File: pythonsnippets/primepairs/primepairs.py
#quick python example#
#primepairs.py: find the pairs of prime numbers#
#between a minimum value and a maximum value   #


range_min = input("Enter in a minimum value: ")
range_max = input("Enter in a maximum value: ")

#isa_prime_num: Int -> Boolean
#is this a prime number?
def isa_prime_num(num):
    if (num==0)|(num==1):
        return False #we don't consider 0|1 to be prime
    for n in range (2,(num-1)):
        if ((num%n) == 0):
            return False
    else:
        return True


for n in range (range_min,range_max):
    if isa_prime_num(n):
        if isa_prime_num(n+2):
            print n," ", (n+2)
